src/us/kbase/common/utils/AweUtils.java

```diff
@@ -65,33 +71,53 @@ public static String runTask(String aweServerUrl, String pipeline,
         builder.addBinaryBody("upload", buffer.toByteArray(),
                 ContentType.APPLICATION_OCTET_STREAM, "tempjob.json");
         httpPost.setEntity(builder.build());
-        HttpResponse response = httpClient.execute(httpPost);
-        String postResponse = EntityUtils.toString(response.getEntity());
-        Map<String, Object> respObj = mapper.readValue(postResponse, Map.class);
+        Map<String, Object> respObj = parseAweResponse(httpClient.execute(httpPost));
         Map<String, Object> respData = (Map<String, Object>)respObj.get("data");
         if (respData == null)
             throw new IllegalStateException("AWE error: " + respObj.get("error"));
         String aweJobId = (String)respData.get("id");
         return aweJobId;
     }
 
+    @SuppressWarnings("unchecked")
+    public static Map<String, Object> parseAweResponse(
+            HttpResponse response) throws IOException, ClientProtocolException,
+            JsonParseException, JsonMappingException, AweResponseException {
+        String postResponse = EntityUtils.toString(response.getEntity());
+        Map<String, Object> respObj = new ObjectMapper().readValue(postResponse, 
+                Map.class);
+        int status = response.getStatusLine().getStatusCode();
+        Integer jsonStatus = (Integer)respObj.get("status");
+        Object errorObj = respObj.get("error");
+        if (status != 200 || jsonStatus != null && jsonStatus != 200 ||
+                errorObj != null) {
+            if (jsonStatus == null)
+                jsonStatus = status;
+            Object message = respObj.get("error");
```

Comment:
For shock the error contents are always a List<String> with one entry, assuming AWE is the same I'd just pull out the string and use that instead of storing the whole object in the exception